### PR TITLE
[MM-60073] Migrate `/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx` from class to function component 

### PR DESCRIPTION
**Pull request by Tihomir-N**
#### Summary
- Refactor `/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx` from class component to functional component.
- Refactor component`s tests using react-testing-library

**QA Notes:**
This is a small settings component located in System Console at Authentication > Guest Access ([follow instructions](https://docs.mattermost.com/configure/authentication-configuration-settings.html#guest-enable) to view). There are no functional changes.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost/issues/27895
Jira https://mattermost.atlassian.net/browse/MM-60073

#### Screenshots
No UI changes

#### Release Note
NONE

### CODE REVIEW

File: webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx [hunk 1]
```diff
@@ -31,61 +39,59 @@ export default class CustomEnableDisableGuestAccountsSetting extends React.PureC
...
+                        defaultMessage='Disabling guest access will revoke all current Guest Account sessions. Guests will no longer be able to login and new guests cannot be invited into Mattermost. Guest users will be marked as inactive in user lists. Enabling this feature will not reinstate previous guest accounts. Are you sure you wish to remove these users?'
+                    />
+                }
+                confirmButtonText={
+                    <FormattedMessage
+                        id='admin.guest_access.disableConfirmButton'
+                        defaultMessage='Save and Disable Guest Access'
+                    />
+                }
+                onConfirm={() => handleChange(id, false, true)}
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx:90]: This generates a new function on every render. Can you move it to a `useCallback`?

File: webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx [hunk 2]
```diff
@@ -19,8 +19,16 @@ type Props = {
...
+const CustomEnableDisableGuestAccountsSetting: React.FC<Props> = ({
+    id,
+    value,
+    onChange,
+    cancelSubmit,
+    disabled,
+    setByEnv,
+    showConfirm,
+}) => {
+    const handleChange = useCallback((id: string, value: boolean, submit = false) => {
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx:31]: Not sure if the linter will complain, but you are redefining here several variables (id and value) which makes things a bit more confusing. Do you mind renaming the variables?

I guess they could be `targetId`, and `newValue`, but I am horrible at naming...

File: webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx [hunk 3]
```diff
@@ -19,8 +19,16 @@ type Props = {
     showConfirm: boolean;
 }
 
-export default class CustomEnableDisableGuestAccountsSetting extends React.PureComponent<Props> {
-    public handleChange = (id: string, value: boolean, submit = false) => {
+const CustomEnableDisableGuestAccountsSetting: React.FC<Props> = ({
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.tsx:22]: Just for consistency with most of the rest of the code, there is no need to be explicit with the React.FC type. We can leave it something like `const foo = ({...}: Props) {`
```suggestion
const CustomEnableDisableGuestAccountsSetting = ({
```

File: webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.test.tsx [hunk 4]
```diff
@@ -18,68 +21,70 @@ describe('components/AdminConsole/CustomEnableDisableGuestAccountsSetting', () =
         showConfirm: false,
     };
 
+    const intlProviderProps = {
+        defaultLocale: 'en',
+        locale: 'en',
+    };
+
+    const renderComponent = (props = {}) => {
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.test.tsx:29]: You should be able to use `renderWithContext`, that does most of what you are doing here.

File: webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.test.tsx [hunk 5]
```diff
@@ -18,68 +21,70 @@ describe('components/AdminConsole/CustomEnableDisableGuestAccountsSetting', () =
...
-                value: true,
-            };
-
-            const wrapper = shallow(
-                <CustomEnableDisableGuestAccountsSetting {...props}/>,
-            );
+    describe('renders correctly', () => {
+        test('when enabled', () => {
+            const wrapper = renderComponent({value: true});
             expect(wrapper).toMatchSnapshot();
```
Comment (larkox) [inline webapp/channels/src/components/admin_console/custom_enable_disable_guest_accounts_setting.test.tsx:51]: What you want to match to the snapshot is `wrapper.container`, not the whole `wrapper`. The reality is that what `render` returns is not really a wrapper, but an object with several util functions. That is why the snapshot looks so weird.

**Review by larkox — CHANGES_REQUESTED** [review]
Great work! Specially with the tests. A couple of comments though.

**Comment by Tihomir-N** [comment]
@larkox Thanks for the comments. I have applied the changes. Let me know, if anything is needed.

**Review by larkox — APPROVED** [review]
Looks great. Thanks!

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Comment by M-ZubairAhmed** [comment]
@Tihomir-N can you also resolve the conflicts

**Comment by Tihomir-N** [comment]
@M-ZubairAhmed Merge it's done! 🚀

**Pull request merged**